### DISCUSSION TITLE
Startup Script PIA with Port Forwarding

### DISCUSSION THREAD

**Discussion by b-nicholson** [qdm12/gluetun#616] [Q&A] [answered]
Hi! I'd like to run a shell script after opening a connection with Private Internet Access (using port forwarding).

I was able to get it working with a custom config file, but I was unable to get port forwarding working.  This is the option I passed though to execute the script within the config file:

`script-security 2`

`up /path/to/script.sh`

I was wondering if it is possible to pass an OpenVPN script option to the config file, or using cli while still using the PIA presets? I appreciate any help!

**Comment by qdm12** [comment]
Hey there,

It's not really possible for now.
Note however PIA port forwarding is supported via env variables if that's what you're looking for.

My end goal is to have gluetun as a single binary and the docker image based on `scratch` where no shell exists, so I would rather avoid to let shell scripts in there.

If I may ask, what's your shell script doing?

**Comment by b-nicholson** [comment]
Oh darn, OK that is totally understandable. I'm glad I was not missing something obvious.

The script is passing along the IP of a download server to a private tracker's API. Very very basic curl type of deal.

**Comment by qdm12** [comment, marked as answer]
Actually, last commit https://github.com/qdm12/gluetun/commit/f807f756eb6ba92d1653c4e73c487e76d1bb9702 allows for `up` and `down` options, because, well screw it 😄 Just make sure to check out the updated Wiki page: https://github.com/qdm12/gluetun/wiki/Openvpn-file

I created https://github.com/qdm12/gluetun/issues/620 for running custom commands when the tunnel is up (and when it stops) such that it works for both OpenVPN and Wireguard and will be able to work with a single binary.

**Comment by panomitrius** [comment]
It's unclear to me from the above what parameters to pass when creating the docker container to launch a custom script upon VPN connection. I've been looking through the Wiki page but can't find anything there.

**Reply by qdm12** [reply]
You can bind mount `/yourhostscript.sh:/inthecontainerscript.sh` and use `up /inthecontainerscript.sh` in a custom Openvpn configuration file, see the 'custom' provider in the wiki.

**Comment by qdm12** [comment]
> The script is passing along the IP of a download server to a private tracker's API. Very very basic curl type of deal.

If this has to do with port forwarding, you can now use `VPN_PORT_FORWARDING_UP_COMMAND`, see https://github.com/qdm12/gluetun-wiki/blob/main/setup/advanced/vpn-port-forwarding.md#custom-port-forwarding-updown-command
